packages/react-magma-dom/src/components/IndeterminateCheckbox/index.tsx

```diff
@@ -159,37 +170,45 @@ export const IndeterminateCheckbox = React.forwardRef<
           ref={ref}
           type="checkbox"
           onChange={handleChange}
+          tabIndex={-1}
         />
-        <StyledLabel htmlFor={id} isInverse={isInverse} style={labelStyle}>
-          <StyledFakeInput
-            isChecked={isChecked}
-            color={color}
-            disabled={disabled}
-            hasError={hasError}
-            hideFocus={props.hideFocus}
-            isIndeterminate={isIndeterminate}
-            isInverse={isInverse}
-            style={inputStyle}
-            theme={theme}
-            aria-hidden="true"
-          >
-            {isIndeterminate ? (
-              <IndeterminateCheckBoxIcon
-                testId="indeterminateIcon"
-                size={theme.iconSizes.medium}
-              />
-            ) : isChecked ? (
-              <CheckBoxIcon size={theme.iconSizes.medium} />
+        <div
+          tabIndex={!disabled && 0}
+          role="checkbox"
+          aria-checked="mixed"
```

Comment:
When we add this, we now have an input with aria-checked mixed AND a div with it. Is this okay? How do other systems handle this?

Reply:
Also, this mixed status is hardcoded so even when it becomes checked, `mixed` is still in the dom.